content/blog/2022-06-28-serving-models-with-mlem.md

```diff
@@ -0,0 +1,277 @@
+[here](https://github.com/iterative/stale-model-example/tree/mlem-serve).
+
+## Train a model
+
+There are instructions in the project
+[README](https://github.com/iterative/stale-model-example/tree/mlem-serve#readme)
+on how to get everything you need installed and running. Make sure you run a few
+experiments with:
+
+```dvc
+$ dvc exp run
+```
+
+That way you can see what it's like actually training a model with DVC. We use
+some bicycle data to make a prediction model that helps end users know how many
+bikes to place around an area of the city. After you've run a few experiments,
+take a look at the metrics table:
+
+```dvc
+$ dvc exp show --drop 'Created|./clf|./data|src/evaluate.py|./src/train.py|./models/model.pkl|./data/train.pkl'
+```
+
+_NOTE:_ We're dropping a lot of columns to simplify the data shown in the table.
+
+```dvctable
+  neutral:**Experiment**                metric:**avg_prec**   metric:**roc_auc**   param:**train.seed**   param:**train.n_est**   param:**train.min_split**
+ ───────────────────────────────────────────────────────────────────────────────────────────
+  workspace                  0.72417   0.38926   20210428     250           75
+  mlem-serve                 0.76681   0.38867   20210428     300           75
+  ├── c8d952b [exp-4b026]    0.72417   0.38926   20210428     250           75
+  ├── da78e7c [exp-7a2e2]    0.74454    0.5528   20210428     450           75
+  ├── b39470f [exp-5ebf6]    0.42526   0.82722   20210428     400           28
+  ├── 03c6d57 [exp-f8c3a]    0.74454    0.5528   20210428     400           75
+  ├── 75b1b3c [exp-2ba75]    0.67913   0.38207   20210428     350           75
+  └── 40b0075 [exp-e0dc1]    0.76681   0.38867   20210428     300           75
+ ───────────────────────────────────────────────────────────────────────────────────────────
+```
+
+Of course you would run many more experiments in order to determine the best
+model for production, but in this example, the metrics for `exp-5ebf6` look
+pretty good. So we'll take this model and get it ready to serve to production.
+
+## Save the model
+
+First, we need to install the `MLEM` package with:
+
+```dvc
+$ pip install 'mlem[fastapi]'
+```
+
+Then, inside of the `train.py` script, we need to add the MLEM import to save
+the models as we experiment.
+
+```python
+# train.py
+
+import os
+import pickle5 as pickle
+import sys
+import yaml
+
+from mlem.api import save
+import numpy as np
+from sklearn.ensemble import RandomForestClassifier
+...
+```
+
+Then you can add the `save` function to the end of the training script.
+
+```python
+# train.py
+...
+clf = RandomForestClassifier(
+    n_estimators=n_est, min_samples_split=min_split, n_jobs=2, random_state=seed
+)
+
+clf.fit(x, labels)
+
+save(
+    clf,
+    "clf",
+    sample_data=x,
+    description="Random Forest Classifier",
+)
+```
+
+_Note:_ If you want to see what is happening in the DVC pipeline, check out the
+`dvc.yaml` file in the project.
```

Comment:
Shat do you mean here? Some step seems to be missing. Are you assuming `exp run` is used after the edit? It's not indicated. (This may be irrelevant if we change the section order as I suggested above.)